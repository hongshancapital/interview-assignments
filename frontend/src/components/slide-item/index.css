.slide-item {
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: relative;
}

.slide-img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: bottom;
  transform: translateY(10%) scale(0.55);
}

.slide-desc {
  position: absolute;
  top: 25%;
  width: 100%;
}
