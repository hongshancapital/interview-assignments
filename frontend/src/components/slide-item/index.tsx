import React from 'react';
import { DescContentConfig } from '../../models';
import './index.css';

interface SlideItemProps {
  /** 图片url */
  imgUrl: string;
  /** 文本描述配置 */
  descContents: DescContentConfig[];
}

export const SlideItem: React.FC<SlideItemProps> = (props: SlideItemProps) => {
  const { imgUrl, descContents = [] } = props;

  /** 渲染背景图 */
  const renderBgImg = () => {
    return imgUrl ? <img className='slide-img' src={imgUrl} alt='slide-img' /> : null;
  }

  /** 渲染文字描述内容 */
  const renderDescContents = () => {
    return (
      <div className='slide-desc' data-testid='slide-desc'>
        {
          descContents.map((desc) => {
            const { id, text, style } = desc;
            return <div key={id} style={{...style}}>{text}</div>;
          })
        }
      </div>
    );
  }

  return (
    <div className='slide-item'>
      {renderBgImg()}
      {renderDescContents()}
    </div>
  );
}
