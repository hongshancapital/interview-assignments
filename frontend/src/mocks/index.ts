import { DEFAULT_ANIMATION_CONFIG } from '../constant';
import { AnimationConfig, SlideItem } from '../models';

export const mockAnimationConfig: AnimationConfig = DEFAULT_ANIMATION_CONFIG;

export const mockSlideItems: SlideItem[] = [
  {
    slideId: 1,
    imgUrl: require('../assets/iphone.png'),
    bgColor: '#111111',
    descContents: [
      {
        id: '1',
        text: 'xPhone',
        style: {
          color: 'white',
          fontSize: '40px',
          fontWeight: 'bold',
          lineHeight: '40px',
        }
      },
      {
        id: '2',
        text: 'Lots to love.Less to spend.',
        style: {
          color: 'white',
          fontSize: '28px',
          fontWeight: 'normal',
          lineHeight: '28px',
          marginTop: '10px',
        }
      },
      {
        id: '3',
        text: 'Starting at $399.',
        style: {
          color: 'white',
          fontSize: '28px',
          fontWeight: 'normal',
          lineHeight: '40px',
        }
      },
    ],
  },
  {
    slideId: 2,
    imgUrl: require('../assets/tablet.png'),
    bgColor: '#fafafa',
    descContents: [
      {
        id: '1',
        text: 'Tablet',
        style: {
          color: 'black',
          fontSize: '40px',
          fontWeight: 'bold',
          lineHeight: '40px',
        }
      },
      {
        id: '2',
        text: 'Just the right amount of everything.',
        style: {
          color: 'black',
          fontSize: '28px',
          fontWeight: 'normal',
          lineHeight: '28px',
          marginTop: '10px',
        }
      },
    ],
  },
  {
    slideId: 3,
    imgUrl: require('../assets/airpods.png'),
    bgColor: '#f1f1f3',
    descContents: [
      {
        id: '1',
        text: 'Buy a Tablet or xPhone for college.',
        style: {
          color: 'black',
          fontSize: '40px',
          fontWeight: 'bold',
          lineHeight: '40px',
        }
      },
      {
        id: '2',
        text: 'Get airPods.',
        style: {
          color: 'black',
          fontSize: '40px',
          fontWeight: 'bold',
          lineHeight: '52px',
        }
      },
    ],
  },
];
