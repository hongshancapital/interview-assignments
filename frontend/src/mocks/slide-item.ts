import { mockSlideItems } from '.';

export const SlideItemProps = mockSlideItems[0];