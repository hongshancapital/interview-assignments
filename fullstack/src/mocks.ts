import { Express } from 'express';
import Redis, { RedisOptions } from 'ioredis-mock';

export default function (app: Express, redisOptions?: RedisOptions) {
  const redis = new Redis(redisOptions);
  app.use((req, res, next) => {
    req.redis = redis;
    next();
  });
  return redis;
}
