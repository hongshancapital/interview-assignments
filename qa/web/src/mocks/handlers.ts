import {rest} from "msw";

export const baseUrl = (path: string) => {
    return new URL(path, 'http://localhost:5000').toString()
}

export const handlers = [
    rest.get(baseUrl('/toolbox'), (req, res, context) => {
        return res(
            context.status(200),
            context.json({
                "message": "success",
                "data": [
                    {
                        "id": 1,
                        "name": "Appium",
                        "tools": "E2E"
                    },
                    {
                        "id": 2,
                        "name": "Jest",
                        "tools": "Jest"
                    }]
            })
        )
    }),

    rest.get(baseUrl('/toolbox/options'), (req, res, context) => {
        return res(
            context.status(200),
            context.json([
                    "E2E",
                    "Unit Test",
                    "Jest",
                    "Selenium"
                ]
            )
        )
    }),

    rest.post<{ name: string, tools: string }>(baseUrl('/toolbox/create'), (req, res, context) => {
        if (req.body.name == null) {
            return res(context.status(400), context.json({"error": "No name specified"}))
        }
        return res(
            context.status(200),
            context.json({"message": "success"}
            )
        )
    })
]