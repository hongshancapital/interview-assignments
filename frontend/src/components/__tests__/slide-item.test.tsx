import React from 'react';
import { render } from '@testing-library/react';
import { SlideItem } from '../slide-item';
import { SlideItemProps } from '../../mocks/slide-item';

describe('test slide-item', () => {
  it('slide-item snapshot', () => {
    // 渲染
    const { container } = render(<SlideItem {...SlideItemProps} />);
    // 快照记录
    expect(container).toMatchSnapshot();
  });

  it('slide-item elements', () => {
    const { getByAltText, getByText, getByTestId } = render(<SlideItem {...SlideItemProps} />);
    const imageElement = getByAltText('slide-img');
    const xPhoneDescElement = getByText('xPhone');
    const slideDescDom = getByTestId('slide-desc');
    expect(imageElement).toBeInTheDocument();
    expect(xPhoneDescElement).toBeInTheDocument();
    expect(slideDescDom.childNodes.length).toBe(SlideItemProps.descContents.length);
  });
});