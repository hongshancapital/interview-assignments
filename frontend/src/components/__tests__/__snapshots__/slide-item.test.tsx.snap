// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test slide-item slide-item snapshot 1`] = `
<div>
  <div
    class="slide-item"
  >
    <img
      alt="slide-img"
      class="slide-img"
      src="iphone.png"
    />
    <div
      class="slide-desc"
      data-testid="slide-desc"
    >
      <div
        style="color: white; font-size: 40px; font-weight: bold; line-height: 60px;"
      >
        xPhone
      </div>
      <div
        style="color: white; font-size: 30px; font-weight: normal; line-height: 40px; margin-top: 15px;"
      >
        Lots to love.Less to spend.
      </div>
      <div
        style="color: white; font-size: 30px; font-weight: normal; line-height: 40px;"
      >
        Starting at $399.
      </div>
    </div>
  </div>
</div>
`;
